### ISSUE TITLE
Make `GroupByDate` use of Localization

### ISSUE THREAD

**Issue by CedricRaison** [gohugoio/hugo#9745]
For example `{{ time.Format "02 January" .Date }}` return a localized string, if my config is in French it will return `04 avril` for the current date.

the GroupByDate doesn't use localization, for example something like this:
 `{{ range (where site.RegularPages "Type" "in" site.Params.mainSections).GroupByDate "January, 2006" -}}` 
will order my post in English. For the current month it will display `April, 2022`. It should be expected to use the same setting as `Time.Format` and return the month in the local format `Avril, 2022`.

**Comment by bep** [comment]
I agree, but the fix isn't obvious. It's doable, but these are methods on the `Pages` slice, so you would have to look in the first page to get the template function namespace (which isn't there at the moment, but it would be ...). Oh well, we'll find a way, thanks for the issue report.
